ui/src/app/views/concept-homepage.tsx

```diff
@@ -159,33 +194,17 @@ export const ConceptHomepage = withCurrentWorkspace()(
     }
 
     componentDidMount() {
-      this.loadDomains();
+      this.loadDomainsAndSurveys();
     }
 
-    async loadDomains() {
+    async loadDomainsAndSurveys() {
       const {namespace, id} = this.props.workspace;
       try {
-        const conceptsCache: ConceptCacheItem[] = [];
-        const conceptDomainCounts: DomainCount[] = [];
-        const resp = await conceptsApi().getDomainInfo(namespace, id);
-        this.setState({conceptDomainList: resp.items});
-        resp.items.forEach((domain) => {
-          conceptsCache.push({
-            domain: domain.domain,
-            items: [],
-            vocabularyList: []
-          });
-          conceptDomainCounts.push({
-            domain: domain.domain,
-            name: domain.name,
-            conceptCount: 0
-          });
-        });
-        this.setState({
-          conceptsCache: conceptsCache,
-          conceptDomainCounts: conceptDomainCounts,
-          selectedDomain: conceptDomainCounts[0],
-        });
+        const [conceptDomainInfo, surveysInfo] = await Promise.all([
+          conceptsApi().getDomainInfo(namespace, id),
+          conceptsApi().getSurveyInfo(namespace, id)]);
+        this.setState({conceptSurveysList: surveysInfo.items});
```

Comment:
I would probably try to put this into the setState called by loadConceptDomainInfo, to go down to a single setState call, rather than two.

Reply:
It may be worth not abstracting the loadConceptDomainInfo into another function.

Reply:
I left it as is because this story is just first step for surveys (was not sure if we would need more logic for survey stories)

Reply:
I am thinking about it i might put it back for this story , however in future if there is any manipulation to survey data we should separate out the logic in two different functions